drivers/hmis/app/models/hmis/user.rb

```diff
@@ -54,6 +57,28 @@ def load_effective_permissions
       send(permission)
     end
 
+    define_method("#{permission}_for?") do |entity|
+      joins_association = nil
+      joins_association = :projects if entity.is_a?(Hmis::Hud::Project)
+      joins_association = :organizations if entity.is_a?(Hmis::Hud::Organization)
+
+      raise "Invalid entity '#{entity.class.name}'" unless joins_association.present?
+      return false unless send("#{permission}?")
+
+      access_group_ids = Hmis::GroupViewableEntity.includes_project(entity).pluck(:access_group_id) if entity.is_a?(Hmis::Hud::Project)
+      access_group_ids = Hmis::GroupViewableEntity.includes_organization(entity).pluck(:access_group_id) if entity.is_a?(Hmis::Hud::Organization)
```

Comment:
As @gigxz we only need project and organization.  Also, can we adjust the syntax to be either an if/else or a case statement?  I'm not finding the many sequential trailing ifs very easy to follow.